lfshook.go

```diff
@@ -1,60 +1,63 @@
-// Package LFShook allows users to write to the logfiles using logrus.
+// Package lfshook is hook for sirupsen/logrus that used for writing the logs to local files.
 package lfshook
 
 import (
 	"fmt"
+	"github.com/sirupsen/logrus"
 	"io"
 	"log"
 	"os"
+	"path/filepath"
 	"reflect"
 	"sync"
-
-	"github.com/sirupsen/logrus"
 )
 
-// We are logging to file, strip colors to make the output more readable
+// We are logging to file, strip colors to make the output more readable.
 var defaultFormatter = &logrus.TextFormatter{DisableColors: true}
 
-// Map for linking a log level to a log file
-// Multiple levels may share a file, but multiple files may not be used for one level
+// PathMap is map for mapping a log level to a file's path.
+// Multiple levels may share a file, but multiple files may not be used for one level.
 type PathMap map[logrus.Level]string
 
-// Alternatively map a log level to an io.Writer
+// WriterMap is map for mapping a log level to an io.Writer.
+// Multiple levels may share a writer, but multiple writers may not be used for one level.
 type WriterMap map[logrus.Level]io.Writer
 
-// Hook to handle writing to local log files.
-type lfsHook struct {
+// LfsHook is a hook to handle writing to local log files.
+type LfsHook struct {
 	paths     PathMap
-	writer    WriterMap
+	writers   WriterMap
 	levels    []logrus.Level
 	lock      *sync.Mutex
 	formatter logrus.Formatter
+
+	defaultPath      string
+	defaultWriter    io.Writer
+	hasDefaultPath   bool
+	hasDefaultWriter bool
 }
 
-// Given a map with keys equal to log levels.
-// We can generate our levels handled on the fly, and write to a specific file for each level.
-// We can also write to the same file for all levels. They just need to be specified.
-func NewHook(levelMap interface{}, userFormatter logrus.Formatter) *lfsHook {
-	var formatter logrus.Formatter
-	if userFormatter != nil {
-		formatter = userFormatter
-	} else {
-		formatter = defaultFormatter
-	}
-	hook := &lfsHook{
-		lock:      new(sync.Mutex),
-		formatter: formatter,
+// NewHook returns new LFS hook.
+// Accepts WriterMap or PathMap in the place of levelMap.
+// If using WriterMap, user is responsible for closing the used io.Writer.
+func NewHook(levelMap interface{}, formatter logrus.Formatter) *LfsHook {
+	hook := &LfsHook{
+		lock: new(sync.Mutex),
 	}
 
+	hook.SetFormatter(formatter)
+
 	switch levelMap.(type) {
+	case nil:
+		break
```

Comment:
If we go this path it might be good to instead support a case for `string` (default path) and a case for `io.Writer` (default writer).

I'm not a big fan of allowing the user to successful create a hook that has no valid output paths configured.

Reply:
Done on [83da246](https://github.com/rifflock/lfshook/pull/25/commits/83da2467e40d480a00e8d24db7018f089d6f6f2a).